cartridges/int_alma/cartridge/controllers/Alma.js

```diff
@@ -263,4 +263,26 @@ server.post('CreatePaymentUrl', server.middleware.https, function (req, res, nex
     return next();
 });
 
+server.post(
+    'Refund',
+    server.middleware.https,
+    function (req, res, next) {
+        var OrderMgr = require('dw/order/OrderMgr');
+        var refundHelper = require('*/cartridge/scripts/helpers/almaRefundHelper');
+        var order = OrderMgr.searchOrder('orderNo={0}', req.querystring.id);
+        var amount = req.querystring.amount;
+
+        try {
+            res.json(refundHelper.refundPaymentForOrder(order, amount));
+        } catch (e) {
+            res.setStatusCode(500);
+            res.json({
+                error: refundHelper.refundPaymentForOrder(order, amount)
```

Comment:
In JS Exceptions doesn't work this way ;)
The exception you have thrown is in the `e` variable ([doc](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error#examples)) .

so maybe instead of calling the function again (which will probably thrown an exception again and produce a uncontrolled error), we could just add e.message ?